xarray/namedarray/core.py

```diff
@@ -0,0 +1,432 @@
+from __future__ import annotations
+
+import copy
+import math
+import typing
+from collections.abc import Hashable, Iterable, Mapping
+
+import numpy as np
+
+# TODO: get rid of this after migrating this class to array API
+from xarray.core import dtypes
+from xarray.core.indexing import ExplicitlyIndexed
+from xarray.core.utils import Default, _default
```

Comment:
these imports hinder the objective of removing all imports of internal Xarray classes and utilities from `namedarray`.  i am curious to hear if anyone has ideas on how to overcome this issue.

Reply:
`dtypes` we may have to vendor (I did the same with flox).

For  the `Default` singleton , can you open an issue please?

for `ExplicitlyIndexed`, I think we handle that by moving those arrays to follow array protocols.

I think this can all be done in a future PR too.